.github/workflows/test.yml

```diff
@@ -47,12 +37,38 @@ jobs:
         pip install wheel
         pip install mypy types-requests types-PyYAML
         pip install .[tests]
+    - name: Remove build artifacts
+      run: rm -rf build dist
+    - name: Load env vars from env.test to GitHub Actions workflow
+      shell: bash
+      run: |
+        env_file="env.test"
+
+        # Load needed env variables from file to the environment
+        set -a
+        source "$env_file"
+        set +a
+
+        # Read each line in the file
+        while IFS= read -r line || [[ -n "$line" ]]; do
+            # Trim leading and trailing whitespace
+            trimmed_line=$(echo "$line" | xargs)
+            # Skip empty lines and lines starting with '#'
+            if [[ -z "$trimmed_line" || "$trimmed_line" =~ ^# ]]; then
+                continue
+            fi
+            # Extract variable name (part before the first '=')
+            variable_name=$(echo "$trimmed_line" | cut -d '=' -f 1)
+
+            # Set the needed env vars in the GitHub Actions workflow
+            echo "$variable_name=${!variable_name}" >> "$GITHUB_ENV"
+        done < "$env_file"
```

Comment:
Why such complication is necessary or why 

```
cat env.test >> "$GITHUB_ENV"
```
is insufficient? ie. have you tried it and it has failed how?

Reply:
No, doing that isn't sufficient.

Here are the reasons.

1. GH workflow doesn't do variable expansion, which is needed in defining some of the variables in `env.test`. I tried a verified this behavior of GH workflow yesterday.
2. According to John's earlier post, all the lines appended to `"$GITHUB_ENV"` must be of the form `varname=value`. Since the `env.test` file contains comments, so appending the content of `env.test` without modification to `"$GITHUB_ENV"` wouldn't work. I just verified this behavior with https://github.com/datalad/datalad-registry/pull/317/commits/c9e52d95c0cf07e8a7dd382957e38a4f4fbd9137 and https://github.com/datalad/datalad-registry/pull/317/commits/38fa0e96b2dfc162d7eb7099cefaf56dccae2e16.

Reply:
ah, I was not aware that it cannot do variables substitution right there -- that's kinda odd... let me think about that.
As for comments -- trivial to address with smth like

```
cat env.testing | sed -e 's, *#.*,,g' | grep -v '^$' >> "$GITHUB_ENV"
```

getting back to above aspect of expansions, with chatgpt I could only master following abomination

```
❯ /bin/bash -c 'declare -p | sort > /tmp/before.env; source env.testing ; declare -p | sort > /tmp/after.env; comm -13 /tmp/before.env /tmp/after.env | grep -v -e "\(_=\|LINES\|COLUMNS\)" | sed -ne "/--/s,.*-- *,,gp"'
CELERY_BROKER_URL="amqp://tester:testpass@localhost:5672"
CELERY_RESULT_BACKEND="redis://localhost:6379"
POSTGRES_DB="pgdb"
POSTGRES_PASSWORD="testpass"
POSTGRES_USER="tester"
RABBITMQ_DEFAULT_PASS="testpass"
RABBITMQ_DEFAULT_USER="tester"
SQLALCHEMY_DATABASE_URI="postgresql+psycopg2://tester:testpass@localhost:5432/pgdb"
```
which I also dislike... 

@jwodder -- do you know/see any pretty way to inherit changes in environment from `. env.testing` across steps or instruct github susbtitute variable names within GITHUB_ENV?

Reply:
@yarikoptic You might be able to use [`python-dotenv`](https://github.com/theskumar/python-dotenv)'s CLI to process `env.testing` into just variable assignments with expansions applied.

Reply:
thanks! added  a note on that to
- https://github.com/datalad/datalad-registry/issues/224#issuecomment-1971688641

for here/now I think we can simplify it a little

```shell
env_file=./env.testing

set -a
. "$env_file"
set +a

sed -e "s, *#.*,,g" < "$env_file" \
| grep -v '^$' \
| while IFS="=" read -r var _; do 
    echo $var="${!var}"; 
done;
```

which produces
```shell
❯ bash /tmp/1.sh
RABBITMQ_DEFAULT_USER=tester
RABBITMQ_DEFAULT_PASS=testpass
POSTGRES_DB=pgdb
POSTGRES_USER=tester
POSTGRES_PASSWORD=testpass
CELERY_BROKER_URL=amqp://tester:testpass@localhost:5672
CELERY_RESULT_BACKEND=redis://localhost:6379
SQLALCHEMY_DATABASE_URI=postgresql+psycopg2://tester:testpass@localhost:5432/pgdb
```

Reply:
> ```shell
> env_file=./env.testing
> 
> set -a
> . "$env_file"
> set +a
> 
> sed -e "s, *#.*,,g" < "$env_file" \
> | grep -v '^$' \
> | while IFS="=" read -r var _; do 
>     echo $var="${!var}"; 
> done;
> ```

That's great. However, since we are going to use python-dotenv in our project for other purposes as well (good idea). We can leverage the CLI of python-dotenv as @jwodder recommended earlier to simplify the `test.yml` file even further (see my latest commits).

Reply:
oh cool -- I guess I have missed that it could be used that way!